### PR TITLE
@group setup

### PR DESCRIPTION
**Pull request by AlexChetverov**
Added test for activation rewards

### CODE REVIEW

File: test/sequential/xyk-pallet.API.bootstrap.claimAndActivate.test.ts [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+  bootstrapCurrency = await Assets.issueAssetToUser(sudo, toBN("1", 20), sudo);
+
+  testUser1 = new User(keyring);
+  keyring.addPair(testUser1.keyRingPair);
+
+  testUser2 = new User(keyring);
+  keyring.addPair(testUser2.keyRingPair);
+});
+
+test("bootstrap - GIVEN a user who provisioned  AND  bootstrap is finished WHEN pool was  promoted THEN claim and activate successfully works AND rewards are activated", async () => {
```
Comment (goncer) [inline test/sequential/xyk-pallet.API.bootstrap.claimAndActivate.test.ts:87]: lets try to find something about all this ducplicated code :)

**Review by goncer — CHANGES_REQUESTED** [review]

File: test/sequential/bootstrap.claimAndActivate.test.ts [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+  ({ promoting, vesting }) => {
+    beforeEach(async () => {
+      const api = getApi();
+      bootstrapPhase = await api.query.bootstrap.phase();
+      if (bootstrapPhase.toString() === "Finished") {
+        const bootstrapFinalize = await finalizeBootstrap(sudo);
+        eventResponse = getEventResultFromMangataTx(bootstrapFinalize);
+        expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+
+        await waitForNBlocks(4);
```
Comment (goncer) [inline test/sequential/bootstrap.claimAndActivate.test.ts:214]: why waiting 4 blocks?

File: test/sequential/bootstrap.claimAndActivate.test.ts [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+  keyring = new Keyring({ type: "sr25519" });
+
+  sudo = new User(keyring, sudoUserName);
+  keyring.addPair(sudo.keyRingPair);
+
+  await sudo.addMGATokens(sudo);
+});
+
+describe.each`
```
Comment (goncer) [inline test/sequential/bootstrap.claimAndActivate.test.ts:198]: niice!

File: test/sequential/bootstrap.claimAndActivate.test.ts [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+      testUser1,
+      bootstrapBlockNumber
+    );
+    eventResponse = getEventResultFromMangataTx(vestingUser);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  }
+  await testUser1.addMGATokens(sudo);
+  await sudo.mint(bootstrapCurrency, testUser1, toBN("1", 20));
+  await testUser2.addMGATokens(sudo);
+  await sudo.mint(bootstrapCurrency, testUser2, toBN("1", 20));
```
Comment (goncer) [inline test/sequential/bootstrap.claimAndActivate.test.ts:74]: we can save some time using this parallelization  @marian created. 
https://github.com/mangata-finance/mangata-e2e/blob/main/test/sequential/xyk-pallet.API.fees.v2.test.ts#L41
It's about minting mangatas and this is a sequential test, so maybe we can parallelize this !

File: test/sequential/bootstrap.claimAndActivate.test.ts [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+      const bootstrapEventResult = await getEventResultFromMangataTx(
+        bootstrapCurrencyIssue,
+        ["tokens", "Issued", sudo.keyRingPair.address]
+      );
+      const bootstrapAssetId = bootstrapEventResult.data[0].split(",").join("");
+      bootstrapCurrency = new BN(bootstrapAssetId);
+      // bootstrapCurrency = await getNextAssetId();
+      //testUser1.addAsset(bootstrapCurrency);
+
+      [testUser1, testUser2] = setupUsers();
```
Comment (AlexChetverov) [inline test/sequential/bootstrap.claimAndActivate.test.ts:245]: We need to use this function to set up User. It has EQUAL code for sudoUser and User that we used, but if I write it here as text, it doesn't work. I didn't understand why.

File: utils/utils.ts [hunk 6]
```diff
@@ -219,6 +219,29 @@ export const waitForNBlocks = async (n: number) => {
...
+  maxNumberBlocks: number
+) {
+  const lastBlock = (await getBlockNumber()) + maxNumberBlocks;
+  let currentBlock = await getBlockNumber();
+  const api = await getApi();
+  let bootstrapPhase = await api.query.bootstrap.phase();
+  testLog.getLog().info("Waiting for boostrap to be " + bootstrapStatus);
+  while (
+    lastBlock > currentBlock &&
+    bootstrapPhase.toString() !== bootstrapStatus
```
Comment (AlexChetverov) [inline utils/utils.ts:233]: We need to use  bootstrapPhase.toString().

File: test/sequential/bootstrap.claimAndActivate.test.ts (hunk 5 above)
Comment (goncer) [reply test/sequential/bootstrap.claimAndActivate.test.ts:237]: maybe some missing initialization for API or something.  If works, leave this like it is for now :)

File: utils/tx.ts [hunk 7]
```diff
@@ -444,6 +444,17 @@ export const createPool = async (
   return result;
 };
 
+export const promotePool = async (sudoAccount: KeyringPair, liqAssetId: BN) => {
+  testLog.getLog().info(`Promoting pool :${liqAssetId}`);
+  const mangata = await getMangataInstance();
+  const api = await mangata.getApi();
+  const result = await signSendAndWaitToFinishTx(
```
Comment (goncer) [inline utils/tx.ts:451]: careful! 
Sudo operations must be handled with nounce Db!

Comment (goncer) [reply utils/tx.ts:451]: like this:
https://github.com/mangata-finance/mangata-e2e/pull/154/files#diff-c7b01ba73caeee9e0c424b9cbc65030d22db0d866a0a4254bb1f60db624c06b5R975

**Review by goncer — CHANGES_REQUESTED** [review]
only one thing needs to be fixed. 
sudo operation without nonce DB

**Review by goncer — APPROVED** [review]

**Review by goncer — APPROVED** [review]

**Pull request merged**